gbasis/density.py

```diff
@@ -0,0 +1,44 @@
+"""Density Evaluation."""
+import numpy as np
+
+
+def eval_density(charge_density, cart_cont_orb):
+    """Return the evaluation of the density at different points.
+
+    Parameters
+    ----------
+    charge_density : np.ndarray(L,L,)
+        Charge density matrix.
+    cart_cont_orb : np.ndarray(L,N,)
+        Array of cartesian contracted orbitals at different grid points.
+
+    Returns
+    -------
+    density : np.ndarray(N,)
+        Density value at different grid points.
+
+    Raises
+    ------
+    TypeError
+        If cart_cont_orb is not an array of floats.
+        If charge_density is not an array of floats.
+    ValueError
+        If charge_density is not a 2-dimensional matrix.
+
+    """
+    # test that inputs have the correct shape and type
+    if not (isinstance(cart_cont_orb, np.ndarray) and cart_cont_orb.dtype == float):
+        raise TypeError("Contracted cartesian orbitals must be a numpy array of data type float.")
+    if not (isinstance(charge_density, np.ndarray) and charge_density.dtype == float):
+        raise TypeError("Charge density matrix must be a numpy array of data type float.")
+    if (len(charge_density.shape) != 2 or len(charge_density) != len(charge_density[0])):
+        raise ValueError("Charge density matrix must be a square matrix.")
+
```

Comment:
you should also check that the `charge_density.shape[0]` and `cart_cont_orb.shape[0]` have the same size.

Reply:
done.